src/Phuong.eShop.WebApp/components/cart/cart-store.ts

```diff
@@ -0,0 +1,32 @@
+import { CatalogItemDetailModel } from "@/models/catalog-item";
+import { proxy, useSnapshot } from "valtio";
+
+export interface CatalogItemStoreDetailModel extends CatalogItemDetailModel {
+    quantity: number,
+}
+
+export const store = proxy({
+    itemList: [] as CatalogItemStoreDetailModel[],
+});
+export const actions = {
```

Comment:
this need to be refactor like this 
https://valtio.dev/docs/how-tos/how-to-organize-actions
`import { proxy } from 'valtio'

export const state = proxy({
  count: 0,
  name: 'foo',
})

export const inc = () => {
  ++state.count
}

export const setName = (name) => {
  state.name = name
}`